app/javascript/application.js

```diff
@@ -1,8 +1,9 @@
 // Configure your import map in config/importmap.rb. Read more: https://github.com/rails/importmap-rails
 import "@hotwired/turbo-rails";
 import "controllers";
-import * as ActiveStorage from "@rails/activestorage"
-ActiveStorage.start()
+import "./turbo_streams";
```

Comment:
**thought:** Can you update this functionality to be the same as you did for the transfer samples?  That did not require any custom turbo actions.

Reply:
I don't know if I can effectively and thoroughly explain the benefits of the custom turbo_stream action in this PR. My preference is to do a Teams call to do a quick demo and talk it over. I'll try...

Here is an example to test the new custom turbo_stream action functionality:
1. Login to irida-next.
1. Go to a project that the user is at least a maintainer of.
1. Click the samples tab.
1. Select some samples you wish to clone to another project.
1. Click the `clone samples` button.
1. Select a project from the drop down.
1. Click the `submit` button.
1. Select the same samples as before and some other samples.
1. Click the `clone samples` button.
1. Select the same project as before from the drop down.
1. Verify the successful samples are unchecked and the unsuccessful samples are checked.

Transfer samples does not uncheck successful transferred samples during a partial response. Currently, it clears the entire key from storage. This functionality is missing from transfer samples. I tried implementing it in the transfer samples PR, but I went down the wrong path using the form turbo:submit-end event. I want to be able to update transfer samples to include the custom turbo_stream action too.

If a custom turbo_stream is overkill, maybe we could use the storage actions from turbo_power, https://github.com/marcoroth/turbo_power? I didn't want to install a new gem and they don't have an action with this exact functionality.

Lastly, we send the cloned sample ids back in the response, but we don't use the returned sample ids. Now we can use the sample ids to persist the state after cloning/transferring is completed.

Reply:
I definitely understand the benefits of the custom turbo stream action, but I guess my question is, why are we keeping samples selected that failed to clone? The user receives an error message with the samples that failed to clone so I don't see the benefit of keeping them selected. If the idea is that the user can then try to clone those samples to a different project, well in that case the user should have to make a conscious choice about selecting which samples they wanted to clone. 

I feel like we had the same discussion about the transfers as well where originally they weren't being cleared but we updated it to just delete the storage key from session storage

Reply:
Recap from Thursday meeting:
- A user could select 100+ samples and try to clone them to a project but accidentally select the wrong project.
It might be tedious for the user to reselect the same samples if the selected project was, for example, one character off.
 ex.) User selects project "/yersinia/yersinia-pseudotuberculosis/outbreak-2023", but meant to select "/yersinia/yersinia-pseudotuberculosis/outbreak-2022".
- Going to wait for @ericenns to be back to have a discussion.

Reply:
Without using a custom turbo stream action, one could just update their turbo stream response to append an HTML element to the page with data attributes for samples that need to selected or need to be unselected. Then the selection controller could process the data attributes to select or unselect samples.

This achieves the same goal as the custom turbo stream action, but has the benefit of keeping all the selection specific javascript code within a single hotwire controller. This would lead to easier maintenance and understanding.

I'm not sure where I stand yet on the idea of maintaining a partial selection on errors yet, I'll have to think about this.

Reply:
Removed in 6394b9c3ce8973dc00802ab5468aa437769834ce.